numpy/_core/_methods.py

```diff
@@ -132,19 +132,11 @@ def _mean(a, axis=None, dtype=None, out=None, keepdims=False, *, where=True):
             dtype = mu.dtype('f4')
             is_float16_result = True
 
-    ret = umr_sum(arr, axis, dtype, out, keepdims, where=where)
-    if isinstance(ret, mu.ndarray):
-        ret = um.true_divide(
-                ret, rcount, out=ret, casting='unsafe', subok=False)
-        if is_float16_result and out is None:
-            ret = arr.dtype.type(ret)
-    elif hasattr(ret, 'dtype'):
-        if is_float16_result:
-            ret = arr.dtype.type(ret / rcount)
-        else:
-            ret = ret.dtype.type(ret / rcount)
-    else:
-        ret = ret / rcount
+    ret = umr_sum(arr, axis, dtype, ... if out is None else out, keepdims, where=where)
+    # TODO: dtype=type(ret.dtype) castst first and isn't quite ideal that way.
+    ret = um.true_divide(ret, rcount, out=out, dtype=type(ret.dtype), casting='unsafe')
```

Comment:
This should be `out=ret` by just following the array stanza of the above. Is the `dtype` then still needed?